### PR TITLE
Bug Fixes and Add Test Code

### PR DESCRIPTION
**Pull request by jxguan**

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -143,7 +143,33 @@ func (c *Client) getFile(docID int) error {
 func (c *Client) SearchWord(word string) ([]string, float64, error) {
 	possibleDocs := c.server.SearchWord(c.indexer.ComputeTrapdoors(word))
 	args := make([]string, len(possibleDocs)+2)
-	args[0] = "-lZ"
+	args[0] = "-lZw"
```
Comment (strib) [inline client/client.go:144]: Everything from here down is copied code -- you should refactor into a helper function that takes `possibleDocs`.

File: client/client.go [hunk 2]
```diff
@@ -143,7 +143,33 @@ func (c *Client) getFile(docID int) error {
...
+	}
+	return filenames, float64(len(possibleDocs)-len(filenames)) / float64(len(c.lookupTable)-len(filenames)), nil
+}
+
+// SearchWordNaive behaves the same as `SearchWord`, except that it simply
+// downloads all the documents and performs a local search on all the documents.
+func (c *Client) SearchWordNaive(word string) ([]string, float64, error) {
+	possibleDocs := make([]int, len(c.lookupTable))
+	for i := 0; i < len(c.lookupTable); i++ {
+		possibleDocs[i] = i
```
Comment (strib) [inline client/client.go]: Here you're assuming the doc IDs are sequential.  Might be clearer as:

``` go
possibleDocs := make([]int, 0, len(c.lookupTable))
for docID := range c.lookupTable {
  possibleDocs = append(possibleDocs, docID)
}
```

**Comment by strib** [comment]
Looks good to me mod nits, thanks!

File: client/client.go (hunk 1 above)
Comment (jxguan) [inline client/client.go:144]: Fixed

**Pull request merged**